File: archive/buzzad-benefit-pop/app/src/defaultpop/java/com/buzzvil/buzzad/benefit/popsample/java/MainActivity.java
package com.buzzvil.buzzad.benefit.popsample.java;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.buzzvil.buzzad.benefit.BuzzAdBenefit;
import com.buzzvil.buzzad.benefit.pop.BuzzAdPop;
import com.buzzvil.buzzad.benefit.popsample.BuildConfig;
import com.buzzvil.buzzad.benefit.popsample.R;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = MainActivity.class.getSimpleName();

    private TextView textAppVersion;
    private TextView textSdkSdkVersion;
    private Button popShowButton;
    private Button popUnregisterButton;

    private BroadcastReceiver sessionReadyReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            Log.e("SessionKey", "Session is Ready. Ads can be loaded now.");
            Toast.makeText(MainActivity.this, "Session is Ready. Ads can be loaded now.", Toast.LENGTH_SHORT).show();
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setVersionText();

        BuzzAdBenefit.registerSessionReadyBroadcastReceiver(MainActivity.this, sessionReadyReceiver);

        this.popShowButton = findViewById(R.id.pop_show_button);
        popShowButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showPop();
            }
        });

        this.popUnregisterButton = findViewById(R.id.pop_unregister_button);
        popUnregisterButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                BuzzAdPop.getInstance().deactivate(MainActivity.this);
            }
        });
    }

    private void setVersionText() {
        textAppVersion = findViewById(R.id.text_app_version);
        String strAppVersion = BuildConfig.VERSION_NAME;
        textAppVersion.setText(strAppVersion);
        textSdkSdkVersion = findViewById(R.id.text_sdk_version);
        String strBenefitSdkVersion = BuzzAdBenefit.getSdkVersion();
        textSdkSdkVersion.setText(strBenefitSdkVersion);
    }

    private void showPop() {
        BuzzAdPop.getInstance().show();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        BuzzAdBenefit.unregisterSessionReadyBroadcastReceiver(this, sessionReadyReceiver);
    }
}

File: buzzad-benefit-extpoint/app/src/main/java/com/buzzvil/benefit/extpoint/sample/publisher/MainActivity.kt
package com.buzzvil.benefit.extpoint.sample.publisher

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.buzzvil.benefit.extpoint.sample.publisher.usecase.FeedActivityUseCase
import com.buzzvil.benefit.extpoint.sample.publisher.usecase.FeedFragmentUseCase
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private val feedActivityUseCase = FeedActivityUseCase()
    private val feedFragmentUseCase = FeedFragmentUseCase()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initButtons()
    }

    private fun initButtons() {
        runFeedActivityExample.setOnClickListener {
            feedActivityUseCase(this@MainActivity)
        }

        runFeedFragmentExample.setOnClickListener {
            feedFragmentUseCase(this@MainActivity)
        }
    }
}
